src/main/ruby/truffleruby/core/posix.rb

```diff
@@ -446,6 +446,7 @@ def self.write_string_native(io, string, continue_on_eagain)
               return written
             end
           else
+            raise SignalException.new(:SIGPIPE) if errno == 32
```

Comment:
It should only be done for STDOUT, no other IO:
https://github.com/ruby/ruby/commit/6f28ebd585fba1aff1c9591ced08ed11b68ba9e3#diff-92194f057884b3287a3a6bf84e6e3b2bf433a556b68562799252a091744e7854R13438

And hardcoding 32 is problematic, use `Errno::EPIPE::Errno` instead.